doc/atomvm-memory.md

```diff
@@ -0,0 +1,538 @@
+# AtomVM Memory Management
+
+Like most managed execution environments, AtomVM provides automated memory management for compiled Erlang/Elixer applications that run on the platform, allowing developers to focus on the logic of application programs, instead of the minutiae of managing the allocation and disposal of memory in the process heap of the program.
+
+Because Erlang/Elixer, and the BEAM, specifically, is a shared-nothing, concurrency-based language, AtomVM can manage memory independently, for each unit of concurrency, viz., the Erlang process.  While there is some global state, internally, that AtomVM managed (e.g., to manage all running processes in the system), memory management for each individual process can be performed independently of any other process.
```

Comment:
```suggestion
Because Erlang/Elixir, and the BEAM, specifically, is a shared-nothing, concurrency-based language, AtomVM can manage memory independently, for each unit of concurrency, viz., the Erlang process.  While there is some global state, internally, that AtomVM managed (e.g., to manage all running processes in the system), memory management for each individual process can be performed independently of any other process.
```